topic_manager.go

```diff
@@ -182,7 +182,7 @@ func (m *topicManager) ensureExists(topic string, npar, rfactor int, config map[
 
 		// (or not)
 		if m.topicManagerConfig.NoCreate {
-			return fmt.Errorf("topic %s does not exist but the manager is configured with NoCreate, so it will not attempt to create it", topic)
+			return fmt.Errorf("topic `%s` does not exist but the manager is configured with NoCreate, so it will not attempt to create it", topic)
```

Comment:
```go
return topicNotFoundError(topic)
```

with
```go
type topicNotFoundError string

func (e topicNotFoundError) Error() string {
	return fmt.Sprintf("topic '%s' does not exist but the manager is configured with NoCreate, so it will not attempt to create it", string(e))
}
```

Reply:
I don't think it's better, but i also don't think it's worse.
You see the error from the type name. The message itself is IMO secondary.

https://github.com/lovoo/goka/pull/477/commits/160d15d3c3875110dffb3a9f90346c38e30ff430
This is not idiomatic Go. I think it's very ugly.
And usually we don't compare strings, because they're brittle. Instead we compare types, because changing the name of the type changes it everywhere.

Reply:
alright :)